src/pages/home/report/ReportActionCompose.js

```diff
@@ -323,6 +370,43 @@ class ReportActionCompose extends React.Component {
         this.setState({maxLines});
     }
 
+    /**
+     * Replace the code of emoji and update selection
+     * @param {Number} highlightedEmojiIndex
+     */
+    insertSelectedEmoji(highlightedEmojiIndex) {
+        const commentBeforeColon = this.state.value.slice(0, this.state.colonIndex);
+        const emojiCode = this.state.suggestedEmojis[highlightedEmojiIndex].code;
+
+        // Define the regular expression pattern to match a string starting with a colon and ending with a space or newline character
+        const pattern = /^:[^\n\r]+?(?=$|\s)/;
+        const commentAfterColonWithEmojiNameRemoved = this.state.value.slice(this.state.colonIndex).replace(pattern, ' ');
+
+        this.updateComment(`${commentBeforeColon}${emojiCode}${commentAfterColonWithEmojiNameRemoved}`, false);
+
+        this.setState(prevState => ({
+            selection: {
+                start: prevState.colonIndex + 2,
```

Comment:
Can you add a comment about why we add `2` here?

Reply:
Emojis have a length of 2. I have left a comment regarding this.

Reply:
Emojis have a length of 2. I have left a comment regarding this and added `emoji.length` instead.

Reply:
Is that true for all emojis? I think some can have a length of up to 6?

Reply:
I've checked several dozen emoticons and couldn't find any longer than the one I used. However, to ensure that everything works ok, I'll calculate the length of the specific emoticon that I'm using and use that value instead of a hardcoded value.